packages/calcite-components/src/components/stepper/stepper.e2e.ts

```diff
@@ -683,7 +706,136 @@ describe("calcite-stepper", () => {
+        </calcite-stepper-item>
+        <calcite-stepper-item heading="Step 3" id="step-2">
+          <calcite-button id="button3">Click</calcite-button>
+        </calcite-stepper-item>
+      </calcite-stepper>`);
+
+      const [actionStart, actionEnd] = await page.findAll("calcite-stepper >>> calcite-action");
+      const [stepperItem1, stepperItem2] = await page.findAll("calcite-stepper-item");
+
+      expect(await actionStart.isVisible()).toBe(true);
+      expect(await actionEnd.isVisible()).toBe(true);
+      expect(await stepperItem1.isVisible()).toBe(true);
+      expect(await stepperItem2.isVisible()).toBe(false);
+
+      const actionEndId = actionEnd.getAttribute("id");
+      const actionStartId = actionStart.getAttribute("id");
+      await page.keyboard.press("Tab");
+      await page.waitForChanges();
+      expect(await isElementFocused(page, `#${actionEndId}`, { shadowed: true })).toBe(true);
+
+      await page.keyboard.press("Tab");
+      await page.waitForChanges();
+      expect(await isElementFocused(page, `#button1`)).toBe(true);
+
+      await actionEnd.click();
+      await page.waitForChanges();
+      expect(await stepperItem1.isVisible()).toBe(false);
+      expect(await stepperItem2.isVisible()).toBe(true);
+
+      await page.keyboard.press("Tab");
+      await page.waitForChanges();
+      expect(await isElementFocused(page, `#button2`)).toBe(true);
+
+      await page.keyboard.down("Shift");
+      await page.keyboard.press("Tab");
+      await page.keyboard.up("Shift");
+      await page.waitForChanges();
+      expect(await isElementFocused(page, `#${actionEndId}`, { shadowed: true })).toBe(true);
+
+      await page.keyboard.down("Shift");
+      await page.keyboard.press("Tab");
+      await page.keyboard.up("Shift");
+      await page.waitForChanges();
+      expect(await isElementFocused(page, `#${actionStartId}`, { shadowed: true })).toBe(true);
+    });
+
+    it("should emit calciteStepperItemChange on user interaction", async () => {
+      const page = await newE2EPage();
+      await page.setContent(html`<calcite-stepper style="width: 100px">
+        <calcite-stepper-item heading="Step 1" id="step-1" disabled>
+          <div>Step 1 content</div>
+        </calcite-stepper-item>
+        <calcite-stepper-item heading="Step 2" id="step-2">
+          <div>Step 2 content</div>
+        </calcite-stepper-item>
+        <calcite-stepper-item heading="Step 3" id="step-2">
+          <div>Step 3 content</div>
+        </calcite-stepper-item>
+      </calcite-stepper>`);
+
+      const stepper = await page.find("calcite-stepper");
+      const [actionStart, actionEnd] = await page.findAll("calcite-stepper >>> calcite-action");
+      const [stepperItem1, stepperItem2, stepperItem3] = await page.findAll("calcite-stepper-item");
+      const eventSpy = await stepper.spyOnEvent("calciteStepperItemChange");
+
+      expect(await actionStart.isVisible()).toBe(true);
```

Comment:
Do we need to assert on visibility if this test is focusing on event emitting? If so, the test name might need to be renamed. May apply to other tests.